ckanext/spectrum/actions.py

```diff
@@ -61,6 +64,70 @@ def user_create(next_action, context, data_dict):
     return created_user
 
 
+def dataset_tag_patch(context, data_dict):
+    if 'tags' not in data_dict or not isinstance(data_dict['tags'], dict):
+        raise toolkit.ValidationError(toolkit._(
+            "Must specify 'tags' dict of tags for update in form "
+            "{'old_tag_name1': 'new_tag_name1', 'old_tag_name2': 'new_tag_name2'}"))
+
+    package_search_params = {}
+    for key in allowed_package_search_params:
+        if key in data_dict:
+            package_search_params[key] = data_dict[key]
+
+    package_search_params = _restrict_datasets_to_those_with_tags(package_search_params, data_dict['tags'])
+
+    datasets = toolkit.get_action('package_search')(context, package_search_params)
```

Comment:
As for search params - you're right it may break in the future, I was wondering if someone could break something by adding other parameters, but since all are passed to package_search one could break using that action as well. I'll remove that restriction.

Results extraction - yep, right, it simplifies a lot :).